src/category_theory/sites/sieves.lean

```diff
@@ -438,9 +441,24 @@ begin
     exact ⟨_, _, _, presieve.pullback_arrows.mk _ _ hk, pullback.lift_snd _ _ comm⟩ },
 end
 
+/--
+If `R` is a sieve, then the `category_theory.presieve.functor_pullback` of `R` is actually a sieve.
+-/
+def functor_pullback {X : C} (R : sieve (F.obj X)) : sieve X := {
+  arrows := presieve.functor_pullback F R,
+  downward_closed' := λ _ _ f hf g, by {
+    unfold presieve.functor_pullback,
+    rw F.map_comp,
+    exact R.downward_closed hf (F.map g),
+  }
+}
+
+@[simp] lemma functor_pullback_id {X : C} (R : sieve X) : R.functor_pullback (𝟭 _) = R :=
+by { cases R, unfold functor_pullback, congr }
```

Comment:
Instead of `cases/unfold`, you should use `ext` and have simp lemmas about membership of the `functor_pullback` sieve.